File: resources/day20.py
def house(n):
    total = n
    for i in range(1, (n // 2) + 1):
        if n % i == 0:
            total += i
    return 10 * total

def main():
    with open('resources/day20.txt', 'r') as f:
        n = int(f.read().strip())
        part1 = 0
        while house(part1) < n:
            part1 += 1
        print(f'Part 1: {part1}')

if __name__ == '__main__':
    main()

File: src/day20.c
#include <stdio.h>
#include <stdlib.h>

long house(long n) {
    long total = n;
    for (long i = 1; i <= (n / 2); i++) {
        if (n % i == 0) {
            total += i;
        }
    }
    return 10 * total;
}

int main(void) {
    FILE *f = fopen("resources/day20.txt", "r");
    long n;
    fscanf(f, "%ld", &n);
    fclose(f);
    long part1 = 0;
    printf("Looking for %ld...\n", n);
    while (house(part1) < n) {
        part1 += 1;
        if (part1 % 10000 == 0) {
            printf("%ld: %ld\n", part1, house(part1));
        }
    }
    printf("\nPart 1: %ld\n", part1);
    return 0;
}
